artemis-server/src/main/java/org/apache/activemq/artemis/core/server/impl/ServerSessionImpl.java

```diff
@@ -1185,7 +1182,7 @@ public void deleteQueue(final SimpleString queueToDelete) throws Exception {
       }
 
       if (server.getAddressInfo(unPrefixedQueueName) == null) {
-         targetAddressInfos.remove(queueToDelete);
+         producers.remove(queueToDelete.toString());
```

Comment:
Actually the only time it needs removing is when it is an FQQN, if a producer is sending just to an address it doesnt matter if any of its queues is deleted as in the subscription case. That being said you might have multiple produceres on different sessions sending to the same FQQN so it would need to use a queue delet callback so all producers were removed

Reply:
> Actually the only time it needs removing is when it is an FQQN, if a producer is sending just to an address it doesnt matter if any of its queues is deleted as in the subscription case. 

Ah I guess, so the thinking is it will just try removing something that never actually existed in the sub case.